compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -999,6 +1002,49 @@ public void visit(BLangSimpleVariable varNode) {
         }
     }
 
+    private void analyzeArrayVariableImplicitInitialValue(BLangSimpleVariable varNode) {
+        BLangType varTypeNode = varNode.typeNode;
+        if (varTypeNode == null || varTypeNode.type == null) {
```

Comment:
@SupunS @NatashaWso2 It turned out that for ignore variable (var _) varNode.typeNode is not null yet varNode.typeNode.type is null, hence re-adding the null check.